File: ROS2/day1/hw2/tongsin/src/my_publisher.cpp
#include "my_publisher.hpp"

using namespace std::chrono_literals;

MyPublisher::MyPublisher() : Node("my_publisher")
{
	count_ = 0;
	str_pub_ = this->create_publisher<std_msgs::msg::String>("my_str_topic", 10);
	int_pub_ = this->create_publisher<std_msgs::msg::Int32>("my_int_topic", 10);
	float_pub_ = this->create_publisher<std_msgs::msg::Float32>("my_float_topic", 10);
	bool_pub_ = this->create_publisher<std_msgs::msg::Bool>("my_bool_topic", 10);
	timer_ = this->create_wall_timer(1s, std::bind(&MyPublisher::timer_callback, this));
}

MyPublisher::~MyPublisher()
{
}

void MyPublisher::timer_callback()
{
	auto msg = std_msgs::msg::String();
	auto int_msg = std_msgs::msg::Int32();
	auto float_msg = std_msgs::msg::Float32();
	auto bool_msg = std_msgs::msg::Bool();
	msg.data = "Hello World: " + std::to_string(count_++);
	int_msg.data = count_;
	float_msg.data = static_cast<float>(count_) * 1.1f;
	bool_msg.data = count_ % 2 == 0;
	str_pub_->publish(msg);
	int_pub_->publish(int_msg);
	float_pub_->publish(float_msg);
	bool_pub_->publish(bool_msg);
	RCLCPP_INFO(this->get_logger(), "Published message: '%s'", msg.data.c_str());
}

int main(int argc, char **argv)
{
	rclcpp::init(argc, argv);
	auto node = std::make_shared<MyPublisher>();
	rclcpp::spin(node);
	rclcpp::shutdown();
	return 0;
}

File: ROS2/day1/hw2/tongsin_py/tongsin_py/my_publisher.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String, Int32, Float32, Bool


class MyPublisher(Node):

    def __init__(self):
        super().__init__('my_py_publisher')
        self.str_pub = self.create_publisher(String, 'my_str_topic', 10)
        self.int_pub = self.create_publisher(Int32, 'my_int_topic', 10)
        self.float_pub = self.create_publisher(Float32, 'my_float_topic', 10)
        self.bool_pub = self.create_publisher(Bool, 'my_bool_topic', 10)
        self.timer = self.create_timer(1, self.timer_callback)
        self.count = 0

    def timer_callback(self):
        str_msg = String()
        int_msg = Int32()
        float_msg = Float32()
        bool_msg = Bool()

        self.count += 1
        str_msg.data = 'Hello from Python: {0}'.format(self.count)
        int_msg.data = self.count
        float_msg.data = self.count * 1.1
        bool_msg.data = self.count % 2 == 0

        self.str_pub.publish(str_msg)
        self.int_pub.publish(int_msg)
        self.float_pub.publish(float_msg)
        self.bool_pub.publish(bool_msg)
        self.get_logger().info('Published: {0}'.format(str_msg.data))


def main(args=None):
    rclpy.init(args=args)
    node = MyPublisher()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        node.get_logger().info('Keyboard Interrupt (SIGINT)')
    finally:
        node.destroy_node()
        rclpy.shutdown()


if __name__ == '__main__':
    main()